fixtures/futures/tests/bindings/test_futures.kts

```diff
@@ -203,3 +203,27 @@ runBlocking {
 
     assertApproximateTime(time, 500, "broken sleep")
 }
+
+// Test a future that uses a lock and that is cancelled.
+runBlocking {
+    val job = launch {
+        useSharedResource(100U)
```

Comment:
If the cancellation doesn't work, the `withTimeout` will raise a timeout exception. That's how I've been testing this :-).

Reply:
I'm seeing weird behavior with this.  Instead of seeing any exception, the program just hangs.  I've had slightly better success if I set the timeout in Rust.  The program still hangs, but I'm able to print out something before.